### PR TITLE
Scripts/Borean Tundra: use existing db gossip menu items for Corastrasza

### PR DESCRIPTION
**Pull request by ghost**
**Changes proposed:**
https://wow.gamepedia.com/Corastrasza
- https://wow.gamepedia.com/Aces_High!#Notes
- https://wow.gamepedia.com/Aces_High!_(daily)#Notes

Use existing db gossip menu items for Corastrasza instead of hardcoded text
- remove hardcoded text (deprecated method)
- enum already existing db gossip menu items
- split the quest status check, the quest gossips are different
- reduce redundant whitespace in enum, but keep alignment

**Target branch(es):** 3.3.5/master

- [X] 3.3.5
- [X] master

**Issues addressed:** Closes #  (insert issue tracker number)

Related to https://github.com/aletson/TrinityCore/issues/7 (**Migrate hardcoded gossip texts to database**)

**Tests performed:** (Does it build, tested in-game, etc.)
- builds without issues
- tested in-game

**Known issues and TODO list:** (add/remove lines as needed)

- [x] still open to requests for change

### CODE REVIEW

**Comment by ccrs** [comment]
I'd use `else` on the second `if`, dunnow better be safe right?

**Comment by illfated** [comment]
Thanks, good suggestion. I was a bit tired when I decided to push this, so I didn't focus on that bit.

File: src/server/scripts/Northrend/zone_borean_tundra.cpp [hunk 1]
```diff
@@ -229,15 +229,16 @@ class npc_khunok_the_behemoth : public CreatureScript
 ## npc_corastrasza
 ######*/
 
-#define GOSSIP_ITEM_C_1 "I... I think so..."
-
 enum Corastrasza
 {
-    SPELL_SUMMON_WYRMREST_SKYTALON               = 61240,
-    SPELL_WYRMREST_SKYTALON_RIDE_PERIODIC        = 61244,
+    GOSSIP_ITEM_I_I_TINK_SO                = 0,
```
Comment (jackpoz) [inline src/server/scripts/Northrend/zone_borean_tundra.cpp]: *think

Comment (illfated) [reply src/server/scripts/Northrend/zone_borean_tundra.cpp]: Thanks. I really was tired when I put this together.

**Comment by illfated** [comment]
Anything else that could done to improve the script content (except for verifying by testing)?

**Comment by ccrs** [comment]
yep

**Comment by illfated** [comment]
OK, let's wait until the weekend and see if someone else has got suggestions then.

**Comment by ccrs** [comment]
seems good, if you can test it, we can merge right away

**Comment by illfated** [comment]
OK, just let me level up my character first, from 73 to 80, only by questing ...
(Just kidding) =^_^=

I will mock up a level 80 some time later tonight and test it.

**Comment by illfated** [comment]
First of all: it is imperative to test in-game when making changes in gossip menus and texts.
A lot of that stuff does not come out the way you think it does, unless this is everyday work.

Secondly: I will be needing some more help due to the way the script needed to be expanded.
Corastrasza has got 3 separate `npc_text` gossips and should use all of them, depending on case.
I may have to change the script to use case switching instead of `if - else` and I am a bit rusty.
(also: I have been ill with flu for 10 days now and it seems to turn my brain into porridge)

**Comment by illfated** [comment]
BTW, could someone add the label **[Comp-C++Script]** to this PR?

**Comment by illfated** [comment]
Oh well. Looks like this will take another week to complete.

**Comment by illfated** [comment]
Feels like there is some function which would facilitate a case switch between all 3 gossips, but it will take me forever and a day to find it now that I'm sick. If anyone know or have got an idea which function would be useful instead of those 3 separate if tests, I will be very happy and grateful for a helpful pointer in the right direction.

**Comment by illfated** [comment]
Well, it looks like I was too optimistic for this script to be working now.
I need to modify the conditions for the gossip to be opened when the quest(s) are done.
Completing the first quest, Aces High! (with the current conditions) the gossip does not open.
I definitely need to find the correct conditions for the player to access the reward gossips.

**Comment by illfated** [comment]
Very well, at least it compiles now. Let me test if the gossip works too.

**Comment by illfated** [comment]
Yes, it finally works as intended, with `NPC_TEXT_DO_YOU_POSSESS_THE_COURAGE` and `NPC_TEXT_EAGERLY_AWAITING_YOUR_RETURN` only showing up when the respective quests are taken.

Anyone who would like to argue that the gossip `NPC_TEXT_EAGERLY_AWAITING_YOUR_RETURN` should be used more than only when the daily quest is taken? Feel free to do so, preferably with some proof to indicate when it should be used.

**Comment by illfated** [comment]
I don't have anything else to add to this content.
It is ready to merge whenever you want to do so.

**Comment by illfated** [comment]
if any of the TC members have got time to update the labels, please add the label **[Comp-C++Script]** and replace [Feedback-FixOutdatedMissingWIP] with **[Feedback-PatchFix]**.

File: src/server/scripts/Northrend/zone_borean_tundra.cpp [hunk 2]
```diff
@@ -254,10 +260,20 @@ class npc_corastrasza : public CreatureScript
...
+                SendGossipMenuFor(player, NPC_TEXT_DO_YOU_POSSESS_THE_COURAGE, me->GetGUID());
+            }
+
+            else if (player->GetQuestStatus(QUEST_ACES_HIGH_DAILY) == QUEST_STATUS_INCOMPLETE)
+            {
+                AddGossipItemFor(player, GOSSIP_MENU_ID_CORASTRASZA, GOSSIP_MENU_ITEM_ID_I_AM_READY, GOSSIP_SENDER_MAIN, GOSSIP_ACTION_INFO_DEF + 1);
+                SendGossipMenuFor(player, NPC_TEXT_EAGERLY_AWAITING_YOUR_RETURN, me->GetGUID());
+            }
+
+            else SendGossipMenuFor(player, NPC_TEXT_MOST_DRAGONS_WOULD_FEAR_TO, me->GetGUID());
```
[the remark this answers is not in the document]
Comment (illfated) [reply src/server/scripts/Northrend/zone_borean_tundra.cpp]: OK, I thought `else` was part of the `if - else if - else` statement.

**Comment by illfated** [comment]
I got into some trouble updating the source, had to re-sync my local source and force push the update after a soft reset. I don't know exactly what happened, but it looks like some lines were removed twice, thereby maybe removing more than I should have removed. Do you mind taking another look at the current content and see what needs to be edited now?

**Comment by Aokromes** [comment]
i think this is what i pushed.
https://github.com/TrinityCore/TrinityCore/pull/22833/files#diff-19943f79004547803606689c09dbfed1L258

**Comment by illfated** [comment]
The link does not work for me, it points to one of the original lines of the script:
```diff
-                AddGossipItemFor(player, GOSSIP_ICON_CHAT, GOSSIP_ITEM_C_1, GOSSIP_SENDER_MAIN, GOSSIP_ACTION_INFO_DEF + 1); 
```

**Comment by illfated** [comment]
Sorry, made a mistake when cleaning up my branches.
Will open a new PR tomorrow for the same content.

**Pull request closed without merging**